### PR TITLE
feat: add secret manager library

### PR DESCRIPTION
**Pull request by coryan**
Mostly generated code with these exceptions:

* `BUILD`: add new `experimental-*` rules
* `doc/adding_new_libraries.md` update instructions to further reduce manual steps
* `external/googleapis/update_libraries.sh` updated to maintain proto and dependency lists on future updates to googleapis
* `generator/generator_config.textproto` add new generated library descriptor
* `google/cloud/secretmanager/README.md` light editing to fix link and make sentences less robotic.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7656)

### CODE REVIEW

**Review by scotthart — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7656)* status: 0 of 38 files reviewed, 1 unresolved discussion (waiting on @coryan and @scotthart)

___
*[google/cloud/secretmanager/internal/retry_traits.h, line 29 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7656#-MpDHKMgA-Yi9d_XQ15Q:-MpDHKMgA-Yi9d_XQ15R:b-bxtwd9) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d5c66723fb3093118e6d72e25cc4180559771c0a/google/cloud/secretmanager/internal/retry_traits.h#L29)):*
<details><summary><i alt="summary">Quoted 8 lines of code…</i></summary>

> ```
> struct SecretManagerServiceRetryTraits {
>   static inline bool IsPermanentFailure(google::cloud::Status const& status) {
>     return status.code() != StatusCode::kOk &&
>            status.code() != StatusCode::kInternal &&
>            status.code() != StatusCode::kDeadlineExceeded &&
>            status.code() != StatusCode::kUnavailable;
>   }
> };
> ```
</details>

Was this file handwritten? It doesn't need to be since the retry codes were in the config, internal/secret_manager_retry_traits.h was generated with this struct.

**Review by coryan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7656)* status: 0 of 38 files reviewed, 1 unresolved discussion (waiting on @scotthart)

___
*[google/cloud/secretmanager/internal/retry_traits.h, line 29 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7656#-MpDHKMgA-Yi9d_XQ15Q:-MpDKnHPDwZOMbTbvkm4:b-15z369) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d5c66723fb3093118e6d72e25cc4180559771c0a/google/cloud/secretmanager/internal/retry_traits.h#L29)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

> ```
> struct SecretManagerServiceRetryTraits {
>   static inline bool IsPermanentFailure(google::cloud::Status const& status) {
>     return status.code() != StatusCode::kOk &&
>            status.code() != StatusCode::kInternal &&
>            status.code() != StatusCode::kDeadlineExceeded &&
>            status.code() != StatusCode::kUnavailable;
>   }
> };
> ```
Was this file handwritten? It doesn't need to be since the retry codes were in the config, internal/secret_manager_retry_traits.h was generated with this struct.
</blockquote></details>

Good catch. Yes, I added this file before your changes to automate the retry traits.  Removed.

File: doc/adding_new_libraries.md [hunk 1]
```diff
@@ -9,54 +9,88 @@ which libraries are based on gRPC.
 We will use `tasks` as an example through this document, you will need to
```
Comment (devbww) [inline doc/adding_new_libraries.md:9]: It looks like `tasks` is now only referenced in one place, and perhaps that should be changed to `${library}`.

Comment (coryan) [reply doc/adding_new_libraries.md]: Fixed, thanks.

**Review by scotthart — APPROVED** [review]
> Reviewed 34 of 37 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7656)* status: all files reviewed, 2 unresolved discussions (waiting on @devbww and @scotthart)

**Pull request merged**